sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneEventHubClient.cs

```diff
@@ -170,9 +170,12 @@ internal sealed class TrackOneEventHubClient : TransportEventHubClient
                 ? new BasicRetryPolicy(clientOptions.RetryOptions)
                 : defaultRetryPolicyFactory();
 
-            var client = TrackOne.EventHubClient.Create(endpointBuilder.Uri, eventHubPath, tokenProvider, retryPolicy.CalculateTryTimeout(0), transportType);
+            var operationTimeout = retryPolicy.CalculateTryTimeout(0);
+            var client = TrackOne.EventHubClient.Create(endpointBuilder.Uri, eventHubPath, tokenProvider, operationTimeout, transportType);
+
             client.WebProxy = clientOptions.Proxy;
             client.RetryPolicy = new TrackOneRetryPolicy(retryPolicy);
+            client.ConnectionStringBuilder.OperationTimeout = operationTimeout;
```

Comment:
For some reason, in the track one code, the timeouts are sometimes inherited from the connection string builder rather than the client.